src/flb_utils.c

```diff
@@ -1415,14 +1415,20 @@ int flb_utils_get_machine_id(char **out_id, size_t *out_size)
 
     if (status == ERROR_SUCCESS) {
         *out_id = flb_calloc(1, dwBufSize+1);
+        memcpy(*out_id, buf, dwBufSize);
```

Comment:
please validate the previous memory allocation before doing memcpy()

Reply:
@pwhelan ping

Reply:
@pwhelan ping

Reply:
Done!